src/machine.rs

```diff
@@ -135,6 +135,17 @@ impl MayLeak for MiriMemoryKind {
     }
 }
 
+impl MiriMemoryKind {
+    /// Whether we have a useful allocation span for an allocation of this kind.
+    fn should_save_allocation_span(self) -> bool {
+        use self::MiriMemoryKind::*;
+        match self {
+            Rust | Miri | C | Mmap => true,
```

Comment:
```suggestion
            // Heap allocations are fine since the `Allocation` is created immediately.
            Rust | Miri | C | Mmap => true,
            // Everything else is unclear, let's not show potentially confusing spans.
```